app/controllers/users_controller.rb

```diff
@@ -1,6 +1,39 @@
 class UsersController < ApplicationController
+
+  before_action :authenticate_user!
+
+  # Only users and admins can edit user's profile
+  def edit
+  end
+
+  def show
+     @user = User.find(params[:id])
+  end
+
+  # Only users and admins can update user's profile
+  def update
+    if current_user.update_attributes(user_params)
+      redirect_to edit_user_registration_path
+  end
```

Comment:
Looks like all the other suggestions are fixed.  Still looks like there's, a `def` and `if` and a single `end`, when there should be 2.  I think this should cause problems when triggering the action in the controller.